plugins/modules/cloud/misc/proxmox_kvm.py

```diff
@@ -232,6 +268,13 @@
       - Specifies the VM name. Only used on the configuration web interface.
       - Required only for C(state=present).
     type: str
+  nameserver:
+    description:
+      - 'cloud-init: Sets DNS server IP address.'
+      - Multiple DNS servers can be set seperated by space.
+      - It uses the setting from the host by default if it has not been set.
+    type: str
```

Comment:
I don't like the part "Multiple DNS servers can be set seperated by space" part, I would turn this into a `type: list` and handle the separation in code e.g. `' '.join(param.nameservers)`. also rename it to plural `nameservers`

P.S. ansible has a magic thing where you can pass  `nameservers: 9.9.9.9` and it will be turned into a list  `nameservers = [ '9.9.9.9' ]` if `type=list`

Reply:
the param itself get's passed to the api directly, so renaming would break as it expects `nameserver` (https://pve.proxmox.com/pve-docs/api-viewer/index.html#/nodes/{node}/lxc/{vmid}/config)

Reply:
Like

```
if module.params['nameserver']:
    kwargs['nameserver'] = ' '.join(module.param.nameservers)
```

?

Reply:
The option can be called differently in Ansible than the one the API expects, so calling it `nameservers` is no problem.

Your code fragment for converting looks good! (Except it would be `module.params['nameservers']`, not `module.param.nameservers` :) )